File: app/src/main/java/com/binatestation/kickstart/fragments/dialogs/ProgressDialogFragment.java
package com.binatestation.kickstart.fragments.dialogs;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;

import com.binatestation.kickstart.R;


/**
 * Dialog fragment to show progress dialog.
 */
public class ProgressDialogFragment extends BaseDialogFragment {

    public static final String TAG = "ProgressDialogFragment";

    private static ProgressDialogFragment sProgressDialogFragment;

    private boolean showing;

    public ProgressDialogFragment() {
        // Required empty public constructor
    }

    public static ProgressDialogFragment getInstance() {
        if (sProgressDialogFragment == null) {
            sProgressDialogFragment = newInstance();
        }
        return sProgressDialogFragment;
    }
    /**
     * gets the instance of ProgressDialogFragment
     *
     * @return ProgressDialogFragment
     */
    public static ProgressDialogFragment newInstance() {

        Bundle args = new Bundle();

        ProgressDialogFragment fragment = new ProgressDialogFragment();
        fragment.setArguments(args);
        fragment.setStyle(DialogFragment.STYLE_NO_TITLE, R.style.AppDialogTheme_Translucent);
        fragment.setCancelable(false);
        return fragment;
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        ProgressBar progressBar = new ProgressBar(getContext());
        ViewGroup.LayoutParams layoutParams = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        progressBar.setLayoutParams(layoutParams);
        return progressBar;
    }

    @Override
    public void show(FragmentManager manager, String tag) {
        if (!showing) {
            super.show(manager, tag);
            showing = true;
        }
    }

    @Override
    public void dismiss() {
        if (showing) {
            super.dismiss();
            showing = false;
        }
    }
}

File: kickoff/src/main/java/com/binatestation/android/kickoff/utils/fragments/dialogs/ProgressDialogFragment.kt
/*
 * (c) Binate Station Private Limited. All rights reserved.
 */
package com.binatestation.android.kickoff.utils.fragments.dialogs

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ProgressBar
import androidx.fragment.app.FragmentManager
import com.binatestation.android.kickoff.R

/**
 * Dialog fragment to show progress dialog.
 */
class ProgressDialogFragment : BaseDialogFragment() {

    private var showing: Boolean = false

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment
        val progressBar = ProgressBar(context)
        val layoutParams =
            ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
            )
        progressBar.layoutParams = layoutParams
        return progressBar
    }


    override fun show(manager: FragmentManager, tag: String?) {
        if (!showing) {
            super.show(manager, tag)
            showing = true
        }
    }

    override fun dismiss() {
        if (showing) {
            super.dismiss()
            showing = false
        }
    }

    companion object {

        const val TAG = "ProgressDialogFragment"

        private var sProgressDialogFragment: ProgressDialogFragment? = null

        val instance: ProgressDialogFragment?
            get() {
                if (sProgressDialogFragment == null) {
                    sProgressDialogFragment = newInstance()
                }
                return sProgressDialogFragment
            }

        /**
         * gets the instance of ProgressDialogFragment
         *
         * @return ProgressDialogFragment
         */
        fun newInstance(): ProgressDialogFragment {

            val args = Bundle()

            val fragment = ProgressDialogFragment()
            fragment.arguments = args
            fragment.setStyle(STYLE_NO_TITLE, R.style.AppDialogTheme_Translucent)
            fragment.isCancelable = false
            return fragment
        }
    }
}// Required empty public constructor
